tools/Customization/DevHome.Customization/Views/DevDriveInsightsView.xaml

```diff
@@ -0,0 +1,453 @@
+            </DataTemplate>
+
+            <!-- Listview Template for showing the error message associated with loading DevDrivesListViewModel object. -->
+            <DataTemplate x:Key="DevDriveLoadingErrorTemplate" x:DataType="devDriveModels:DevDrivesListViewModel">
+                <ListView
+                    AutomationProperties.Name="{x:Bind ErrorText, Mode=OneWay}"
+                    HeaderTemplate="{StaticResource DevDriveContainerHeaderTemplate}"
+                    SelectionMode="None"
+                    Margin="0 0 0 10"
+                    ItemContainerStyle="{StaticResource HorizontalCardListViewItemContainerStyle}">
+                    <ListViewItem
+                        IsTabStop="False"
+                        DataContext="{Binding}">
+                        <StackPanel
+                            Margin="0 50 0 50"
+                            HorizontalAlignment="Center"
+                            VerticalAlignment="Center">
+                            <TextBlock
+                                Text="{x:Bind ErrorText}"
+                                Foreground="{ThemeResource TextFillColorPrimaryBrush}"
+                                Margin="0 0 0 10"
+                                HorizontalAlignment="Center"
+                                IsTextSelectionEnabled="True"/>
+                        </StackPanel>
+                    </ListViewItem>
+                </ListView>
+            </DataTemplate>
+
+            <!-- Listview Template for showing the info message when no dev drives were found. -->
+            <DataTemplate x:Key="DevDrivesNotFoundTemplate">
+                <ListView
+                    AutomationProperties.Name="DevDriveInsightsNoDevDrivesFound"
+                    AutomationProperties.LiveSetting="Polite"
+                    HeaderTemplate="{StaticResource DevDriveContainerHeaderTemplate}"
+                    SelectionMode="None"
+                    Margin="0 0 0 10"
+                    ItemContainerStyle="{StaticResource HorizontalCardListViewItemContainerStyle}">
+                    <ListViewItem
+                        IsTabStop="False"
+                        DataContext="{Binding}">
+                        <StackPanel
+                            Margin="0 50 0 50"
+                            HorizontalAlignment="Center"
+                            VerticalAlignment="Center">
+                            <TextBlock
+                                x:Uid="DevDriveInsightsNoDevDrivesFound"
+                                Foreground="{ThemeResource TextFillColorPrimaryBrush}"
+                                FontSize="15"
+                                Margin="0 0 0 10"/>
+                        </StackPanel>
+                    </ListViewItem>
+                </ListView>
+            </DataTemplate>
+        </ResourceDictionary>
+    </UserControl.Resources>
+
+    <ScrollViewer >
+        <StackPanel Orientation="Vertical">

```

Comment:
Why the nested `stackpanel`s?

Reply:
I have an outer stackpanel holding individual stack panels displaying Dev drive cards, optimizer cards and optimized cards respectively.